### PR TITLE
feat(k8s compatibility)

### PR DESCRIPTION
**Pull request by hutchic**
make the plugin more resilient to k8s changes by removing keys we don't care about.

added tests for the three recent k8s version one which is failing but that's unrelated to this plugin

### CODE REVIEW

File: kong/plugins/kubernetes-sidecar-injector/api.lua [hunk 1]
```diff
@@ -37,6 +37,23 @@ local function skip_injection(plugin_config, review_request) -- luacheck: ignore
 end
 
 
+-- Removes unknown fields from the request (i.e. from more recent Kubernetes versions)
+local function remove_unknown_keys(schema, input)
+  local result = {}
+  for k, field in schema:each_field() do
+    local val = input[k]
+    if val then
```
Comment (bungle) [inline kong/plugins/kubernetes-sidecar-injector/api.lua]:
```suggestion
    if val ~= nil then
```

File: kong/plugins/kubernetes-sidecar-injector/api.lua [hunk 2]
```diff
@@ -37,6 +37,23 @@ local function skip_injection(plugin_config, review_request) -- luacheck: ignore
 end
 
 
+-- Removes unknown fields from the request (i.e. from more recent Kubernetes versions)
+local function remove_unknown_keys(schema, input)
+  local result = {}
+  for k, field in schema:each_field() do
```
Comment (bungle) [inline kong/plugins/kubernetes-sidecar-injector/api.lua:48]: I think we should only enter the `for` loop if `type(input) == "table"`.

**Review by bungle — CHANGES_REQUESTED** [review]
Some changes to make it even more robust.

File: kong/plugins/kubernetes-sidecar-injector/api.lua [hunk 3]
```diff
@@ -37,6 +37,23 @@ local function skip_injection(plugin_config, review_request) -- luacheck: ignore
 end
 
 
+-- Removes unknown fields from the request (i.e. from more recent Kubernetes versions)
```
Comment (bungle) [inline kong/plugins/kubernetes-sidecar-injector/api.lua:40]:
```suggestion
-- Removes unknown fields from the request (i.e. from more recent Kubernetes versions)
-- NOTE: this is implemented in Kong 1.3.0 with `schema:process_auto_fields`, so this is here for prior releases
```

File: kong/plugins/kubernetes-sidecar-injector/api.lua (hunk 2 above)
Comment (kikito) [reply kong/plugins/kubernetes-sidecar-injector/api.lua:48]: Done, added an `if` before `local result = {}` which early-exits

**Review by bungle — APPROVED** [review]

**Pull request merged**